src/Models/ModelInstantiator.php

```diff
@@ -331,23 +332,34 @@ public function convertPropertyOnRetrieve(
     }
 
     /**
-     * @param mixed $property
+     * Public for testability.
+     * @param string $property ReflectionProperties->getName() always returns
+     *                         a string which is the key of the array that
+     *                         handles this property.
      * @param array $property_reflection
-     * @return mixed
+     * @return \ReflectionProperty
+     * @throws \TypeError
      */
-    private function getReflectionProperty(mixed $property, array $property_reflection): mixed
+    public function getReflectionProperty(string $property, array $property_reflection): \ReflectionProperty
```

Comment:
We should add the use statement for `ReflectionProperty` instead of using the `\`.